app/controllers/pet_applications_controller.rb

```diff
@@ -1,6 +1,24 @@
 class PetApplicationsController < ApplicationController
+
   def index
     @pet = Pet.find(params[:pet_id])
     @applicants = @pet.adoption_applications
   end
+
+  def update
+    pet = Pet.find(params[:pet_id])
+    pet.update({adoption_status: toggle_adoption_status(pet)})
+    pet_application = PetApplication.where(adoption_application_id: params[:application_id], pet_id: pet.id).first
```

Comment:
This was a little weird, but even though it will only return 1 object I still have to call .first or else it is in the form of an array and I can't access the attributes. Maybe there is a better activerecord method to use besides .where? Let me know if you have any ideas. It's harmless, but a little weird.

Reply:
I'm not sure I understand what is happening here. Would like to go over it, so you can explain in more detail.